### PR TITLE
Remove deprecated identify, createalias and reset

### PR DESCRIPTION
**Pull request by vegaro**
Removal of deprecated methods. Counterpart of https://github.com/RevenueCat/purchases-ios/pull/974/files

[sc-11579]

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- `identify` has been removed
+- `reset` has been removed
+- `createAlias` has been removed
```
Comment (vegaro) [inline MIGRATION.md:3]: created this file to keep track of breaking changes

**Review by taquitos — COMMENTED** [review]
LGTM but I'll wait for somebody with more kotlin experience to 👍 it

File: examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/LoginFragment.kt [hunk 2]
```diff
@@ -22,25 +21,25 @@ class LoginFragment : Fragment() {
         binding.loginButton.setOnClickListener {
             binding.loginUsernameEditText.text?.toString()?.let { userId ->
                 Purchases.sharedInstance
-                    .identifyWith(
+                    .logInWith(
                         userId,
                         { error -> showUserError(requireActivity(), error) },
-                        ::advanceToOverviewFragment
+                        { _, created -> if (created) { advanceToOverviewFragment() } }
```
Comment (aboedo) [inline examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/LoginFragment.kt:27]: this should always go to next fragment, right? Does PurchaseTester differentiate between new users and existing users that logged in?

**Review by aboedo — COMMENTED** [review]
love seeing so much code go

**Review by aboedo — APPROVED** [review]

**Pull request merged**